src/shell/shell.cpp

```diff
@@ -131,47 +133,41 @@ void DOS_Shell::GetRedirection(char *line,
 	*line_write = 0;
 }
 
-bool get_pipe_status(const char *out_file,
+static uint16_t get_output_redirection(const char *out_file,
                      const char *pipe_file,
                      char (&pipe_tempfile)[270],
                      const bool append,
                      bool &failed_pipe)
 {
 	FatAttributeFlags fattr = {};
-	uint16_t dummy = 0;
-	uint16_t dummy2 = 0;
+	uint16_t file_handle = InvalidFileHandle;
 	uint32_t bigdummy = 0;
-	bool status = true;
+	bool success = true;
 	/* Create if not exist. Open if exist. Both in read/write mode */
 	if (!pipe_file && append) {
 		if (DOS_GetFileAttr(out_file, &fattr) && fattr.read_only) {
 			DOS_SetError(DOSERR_ACCESS_DENIED);
-			status = false;
-		} else if ((status = DOS_OpenFile(out_file, OPEN_READWRITE, &dummy))) {
+			success = false;
+		} else if ((success = DOS_OpenFile(out_file, OPEN_READWRITE, &file_handle, true))) {
 			DOS_SeekFile(1, &bigdummy, DOS_SEEK_END);
 		} else {
 			// Create if not exists.
-			status = DOS_CreateFile(out_file,
+			success = DOS_CreateFile(out_file,
 			                        FatAttributeFlags::Archive,
-			                        &dummy);
+			                        &file_handle, true);
```

Comment:
What is `true`? Extract to constant please.

Reply:
Fixed